torch_xla/csrc/ops/generic.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include "ir.h"
+
+namespace torch_xla {
+namespace ir {
+namespace ops {
+
+// Generic IR Node implementation for nodes which can simply be described by a
+// specific OpKind and a lowering function. IR nodes carrying metadata should
+// not be using this class (and have the metadata captured by the LowerFn), but
+// they should instead create a dedicated IR node. Doing the former would limit
+// IR introspection.
+class Generic : public Node {
```

Comment:
Isn't this better named as `Pure`?

Reply:
Pure does not tell me much. OTOH I am not too in love with Generic as well.

Reply:
It's better than `Generic` and just a bit over-general -- there are indeed some operations which are (mathematically) pure but won't use this node, but no in-place operations are going to use it. Unless you have a more precise term, I'd go with it over `Generic`.

Reply:
What does it mean "pure" in this context?
I just do not get how it describes better than "generic".
At least generic conveys the fact that this instance is used for different nodes which only have the lower_fn/generator part differing.
I am OK in changing it, but it needs to land in a better place than generic does.

Reply:
I mean it in the https://en.wikipedia.org/wiki/Pure_function sense, but it sounds like what you actually care about LowerFn being pure, not the operator itself. I do believe the two will overlap substantially in practice, but feel free to keep it as `Generic` (or maybe `Simple`?).